default/generated/spring-framework/spring-framework-5.x-to-6.0-core-container.yaml

```diff
@@ -40,4 +40,53 @@
   links:
     - title: 'Spring 6.0 migration guide'
       url: https://github.com/spring-projects/spring-framework/wiki/Spring-Framework-6.0-Release-Notes#core-container
+      
+- ruleID: spring-framework-5.x-to-6.0-core-container-00020
+  category: potential
+  effort: 3
+  labels:
+  - konveyor.io/source=spring5
+  - konveyor.io/target=spring6+
+  when:
+    java.referenced:
+      pattern: org.springframework.util.concurrent.ListenableFuture*
+      location: IMPORT
+  description: ListenableFuture and related types are deprecated since Spring Framework 6.0.  
+  message: |
+    Spring Framework 6.0 has deprecated support for ListenableFuture and related types (ListenableFutureCallback, SettableListenableFuture, etc.)
+    The reason for deprecation is the introduction of CompletableFuture in Java 8, which offers a richer set of features and better integration with the Java ecosystem.
+    java.util.concurrent.CompletableFuture is now the recommended way to handle asynchronous operations in Spring applications.
+  Example:
```

Comment:
I think this is being taken as a field by the yaml parser - could you maybe substitute this for a sentence and add two spaces to the left so that it's aligned with the rest? It could be something like:
>An example of a migration would be: